sources/web/datalab/polymer/modules/template-manager/template-manager.ts

```diff
@@ -36,44 +34,47 @@ class NotebookTemplate {
   }
 
   /**
-   * Escapes all regex modifier characters in the given string.
+   * Inserts a new cell at position zero that defines python variables
+   * for all of the parameters.
    */
-  private static _regexEscapeAll(s: string) {
-    return s.replace(/([.*+?^${}()|\[\]\/\\])/g, '\\$1');
-  }
-
-  /**
-   * Substitutes all placeholders in the given notebook's cells with their
-   * values.
-   */
-  public populatePlaceholders(notebook: NotebookContent) {
-
-    notebook.cells.forEach((cell: NotebookCell) => {
-      this.parameters.forEach((parameter: TemplateParameter) => {
-        const placeholder = PLACEHOLDER_PREFIX + parameter.placeholder;
-        const escapedPlaceholder = NotebookTemplate._regexEscapeAll(placeholder);
-        const regex = new RegExp(escapedPlaceholder);
-        const value = '\'' + parameter.value.toString() + '\'';
-        cell.source = cell.source.replace(regex, value);
-      });
-    });
-
+  public addParameterCell(notebook: NotebookContent) {
+    const definitionLines = this.parameters.map(parameter =>
+        parameter.placeholder + ' = ' + JSON.stringify(parameter.value));
+    const header = '# Parameter definitions generated by code';
+    const cellText = header + '\n' + definitionLines.join('\n');
+    const newCell: NotebookCell = {
+      cell_type: 'code',
+      execution_count: 0,
+      metadata: {},
+      outputs: [] as string[],
+      source: cellText,
+    } as NotebookCell;
+    notebook.cells.unshift(newCell);  // insert new cell as the first cell
   }
 }
 
 /**
  * This template contains one cell that shows the given table's schema.
  */
-class TableSchemaTemplate extends NotebookTemplate {
-  constructor(tableName: string) {
-    const parameters = [{
-      placeholder: 'TABLE_NAME_PLACEHOLDER',
-      value: tableName,
-    }];
+class BigQueryTableOverviewTemplate extends NotebookTemplate {
+  constructor(dict: { [key: string]: string }) {
```

Comment:
Are string parameters the only type we support? Perhaps we should make this `any`, or one of a defined set of types?

Reply:
Done.